### ISSUE TITLE
JSON Serializing Issue

### ISSUE THREAD

**Issue by Talha-T** [basarat/typescript-collections#91]
Dictionary.toString() gives a lot of unwanted stuff like this:
```json
{
     “table”: {
       “$$o0": {
         “key”: 0,
         “value”: {
           “players”: {
             “table”: {},
             “nElements”: 0
           },
           “id”: 0
         }
       },
       “$$o1”: {
         “key”: 1,
         “value”: {
           “players”: {
             “table”: {
               “$$o895": {
                 “key”: 895,
                 “value”: {
                   “user”: {
                     “displayName”: “895”,
                     “id”: 895,
                     “socketId”: “x-5xlmLI52NNs8jCAAAA”,
                     “state”: 206,
                     “gameId”: 1
                   }
                 }
               },
               “$$o997”: {
                 “key”: 997,
                 “value”: {
                   “user”: {
                     “displayName”: “997",
                     “id”: 997,
                     “socketId”: “yYoKDq1rheXgWi72AAAB”,
                     “state”: 206,
                     “gameId”: 1
                   }
                 }
               }
             },
             “nElements”: 2
           },
           “id”: 1
         }
       }
     },
     “nElements”: 2
   }
}
```
While I am expecting something like this:
```json
{
    "key": "value"
}
```

**Comment by Talha-T** [comment]
Or I noticed that JSON does not call .toString(), is there a way to fix this without calling toString()?

**Comment by Talha-T** [comment]
It looks like JSON also serializes protected properties somehow..

**Comment by episage** [comment]
How about this:
![image](https://user-images.githubusercontent.com/2196095/35773647-69e2fad6-0958-11e8-9adf-774da3139fc3.png)
?

**Comment by Talha-T** [comment]
I need JSON.

**Comment by Talha-T** [comment]
This can be fixed with #92, but no response still

**Comment by episage** [comment]
Closing this because of old age. If the issue persists, discuss it here and/or open a Pull Request please.
